src/services/codefixes/inferFromUsage.ts

```diff
@@ -179,8 +226,10 @@ namespace ts.codefix {
     function annotateSetAccessor(changes: textChanges.ChangeTracker, sourceFile: SourceFile, setAccessorDeclaration: SetAccessorDeclaration, program: Program, host: LanguageServiceHost, cancellationToken: CancellationToken): void {
         const param = firstOrUndefined(setAccessorDeclaration.parameters);
         if (param && isIdentifier(setAccessorDeclaration.name) && isIdentifier(param.name)) {
-            const type = inferTypeForVariableFromUsage(setAccessorDeclaration.name, program, cancellationToken) ||
-                inferTypeForVariableFromUsage(param.name, program, cancellationToken);
+            let type = inferTypeForVariableFromUsage(setAccessorDeclaration.name, program, cancellationToken);
+            if (type === program.getTypeChecker().getAnyType()) {

```

Comment:
Would it be better to use `type.flags & TypeFlags.Any` here?

Reply:
Short answer: no.
Long answer: here we are checking for *exactly* the fallback type produced inside inferFromUsage.ts, not other types that come from the checker like errorType. Those types are already filtered out in addCandidateType, which does use TypeFlags.Any.